### PR TITLE
Align path format for windows.

### PR DESCRIPTION
**Pull request by specter119**
This PR makes the following modifications.

overall:
- change `str(pathlib.Path)` to `pathlib.Path.as_posix()`
- ~convert `pathlib.Path / "a" / "b" / "c"` to pathlib.Path / "a/b/c"~
- replace `parent.parent` with `parents[1]`

for `plugins/lab/fps_lab/utils.py`:
-  use pathlib.Path.rglob to replace glob.glob(pathlib.Path, recursive=True)
- check "jupyterlab" in `package`

---

With these modifications, the following issues on the windows platform solved:
- no labextension is found.
- in the file browser, the path is not separated (displayed as `/ a\b\c /`, and can't enter the parent folder).
- raise error the `package.json` without "jupyterlab"

however, the following errors still exist, and I'm not sure they're windows specific:
- `--root-path` doesn't change the root folder, and the root folder is still `.`
- the `pwd` is always the root folder and doesn't change following the editing file.

---

To sum up, the unix like path is more friendly even in the windows platform, but str(pahtlib.Path) will convert the path as the native format.

### CODE REVIEW

File: plugins/kernels/fps_kernels/routes.py [hunk 1]
```diff
@@ -123,9 +127,9 @@ async def create_session(
     create_session = await request.json()
     kernel_name = create_session["kernel"]["name"]
     kernel_server = KernelServer(
-        kernelspec_path=str(
-            prefix_dir / "share" / "jupyter" / "kernels" / kernel_name / "kernel.json"
-        ),
+        kernelspec_path=(
+            prefix_dir / f"share/jupyter/kernels/{kernel_name}/kernel.json"
+        ).as_posix(),
```
Comment (davidbrochart) [inline plugins/kernels/fps_kernels/routes.py:132]: `kernelspec_path` is used to open a file in the back-end, is this going to work on Windows?

**Review by davidbrochart — COMMENTED** [review]
Thanks a lot @specter119!
I'm wondering if we should use `as_posix()` only when passing paths to the front-end, and not use it or forward slashes (`/`) in the back-end.

**Comment by davidbrochart** [comment]
> * `--root-path` doesn't change the root folder, and the root folder is still `.`
> * the `pwd` is always the root folder and doesn't change following the editing file.

We currently don't handle a root directory (see #236), but we definitely need this.

**Comment by specter119** [comment]
I add these changes to my local jupyverse, and it works as excepted. Together, I will actively use jupyverse for a perioid of time. Since the behavior of jupyter-server (maybe related with tornado) break my daily work, and downgrade to tornado 6.1 doesn't help.

File: plugins/auth/fps_auth/db.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 auth_config = get_config(AuthConfig)
 
-jupyter_dir = Path.home() / ".local" / "share" / "jupyter"
+jupyter_dir = Path.home() / ".local/share/jupyter"
```
[the remark this answers is not in the document]
Comment (davidbrochart) [reply plugins/auth/fps_auth/db.py:21]: From what I can read, I think it works because Python does some tricks to convert forward slashes to backslashes if needed when opening files, but the best practice is still to use `pathlib`'s `/` operator.

File: plugins/kernels/fps_kernels/routes.py (hunk 1 above)
Comment (davidbrochart) [reply plugins/kernels/fps_kernels/routes.py:132]: Same as my [comment above](https://github.com/jupyter-server/jupyverse/pull/243#discussion_r1017558155), I think we should stick to `pathlib`'s `/` operator.

**Comment by davidbrochart** [comment]
Yes, jupyter-server's dependency on Tornado was one of the reasons to start jupyverse.

File: plugins/lab/fps_lab/utils.py [hunk 3]
```diff
@@ -8,9 +7,11 @@ def get_federated_extensions(extensions_dir: Path) -> Tuple[List, List]:
     federated_extensions = []
     disabledExtensions = []
 
-    for path in glob(str(extensions_dir / "**" / "package.json"), recursive=True):
+    for path in extensions_dir.rglob("**/package.json"):
         with open(path) as f:
             package = json.load(f)
+        if "jupyterlab" not in package.keys():
```
Comment (davidbrochart) [inline plugins/lab/fps_lab/utils.py:13]:
```suggestion
        if "jupyterlab" not in package:
```

**Review by davidbrochart — APPROVED** [review]
Thanks Kyle, just a minor change.
